std/conv.d

```diff
@@ -3980,10 +3982,14 @@ private template emplaceImpl(T)
                 else static if (is(Unqual!(ElementEncodingType!Arg) == UE))
                 {
                     import core.stdc.string : memcpy;
-                    assert(N == chunk.length, "Array length missmatch in emplace");
-                    memcpy(cast(void*)&chunk, arg.ptr, T.sizeof);
+                    assert(n == chunk.length, "Array length missmatch in emplace");
+
+                    // This is unsafe as long as the length match is a
+                    // precondition and not an unconditional exception
+                    memcpy(&chunk, arg.ptr, T.sizeof);
```

Comment:
This should really emulate what compiler+druntime does: check the lengths, and throw an ArrayBoundsError otherwise. Note that we can further use `version(!D_NoBoundsChecks)` to also emulate disabled bounds checking...

... would be an idea for a future improvement.